src/hooks/useCalculateSwapPairs.ts

```diff
@@ -25,36 +25,73 @@ const SWAP_TYPES_ORDERED_ASC = [
 ]
 
 type TokenToPoolsMap = {
-  [tokenSymbol: string]: string[]
+  [tokenSymbol: string]: string[] | undefined
+}
+
+export type ExpandedBasicPool = BasicPool & {
+  priceData: ReturnType<typeof getPriceDataForPool>
+  expandedTokens: BasicToken[]
+  expandedUnderlyingTokens: BasicToken[] | null
 }
 
 type TokenToSwapDataMap = { [symbol: string]: SwapData[] }
 export function useCalculateSwapPairs(): (token?: Token) => SwapData[] {
   const [pairCache, setPairCache] = useState<TokenToSwapDataMap>({})
-  const poolsStatuses = usePoolsStatuses()
+  const basicPools = useContext(BasicPoolsContext)
+  const tokens = useContext(TokensContext)
   const { chainId } = useActiveWeb3React()
+  const { tokenPricesUSD } = useSelector((state: AppState) => state.application)
   const [poolsSortedByTVL, tokenToPoolsMapSorted] = useMemo(() => {
-    const sortedPools = Object.values(POOLS_MAP)
-      .filter((pool) => (chainId ? pool.addresses[chainId] : false)) // filter by pools available in the chain
-      .filter((pool) => !poolsStatuses[pool.name]?.isPaused) // paused pools can't swap
-      .sort((a, b) => {
-        const aTVL = poolsStatuses[a.name]?.tvl
-        const bTVL = poolsStatuses[b.name]?.tvl
-        if (aTVL && bTVL) {
-          return aTVL.gt(bTVL) ? -1 : 1
-        }
-        return aTVL ? -1 : 1
+    if (basicPools == null || tokens == null) return []
+    const basicPoolsWithPriceData = (Object.values(basicPools) as BasicPool[])
+      .map((pool) => {
+        const priceData = getPriceDataForPool(tokens, pool, tokenPricesUSD)
+        const expandedTokens = pool.tokens.map((addr) => tokens?.[addr])
+        const expandedUnderlyingTokens =
+          pool.underlyingTokens?.map((addr) => tokens?.[addr]) || null
+        return { ...pool, priceData, expandedTokens, expandedUnderlyingTokens }
       })
-    const tokenToPools = sortedPools.reduce((acc, { name: poolName }) => {
-      const pool = POOLS_MAP[poolName]
+      .filter((pool) => !pool.isPaused) // paused pools can't swap
+      .filter((expandedPool) => {
+        const hasExpandedTokenData = expandedPool.expandedTokens.every(
+          (token) => token != null,
```

Comment:
Because it catches both null and undefined - null is usually explicitly returned, but undefined can come from `object[key]` or `object?.foo.key`. I could probably use an explicit check in more places however, I'll try and make those updates.